platypus/parse.py

```diff
@@ -1,13 +1,96 @@
 # ----------------------------------------------------------------------------
 # Copyright (c) 2015--, platypus development team.
 #
-# Distributed under the terms of the GPL License.
+# Distributed under the terms of the BSD License.
 #
 # The full license is in the file COPYING.txt, distributed with this software.
 # ----------------------------------------------------------------------------
 from __future__ import division
 from itertools import product, izip
-from cogent.parse.blast import MinimalBlastParser9
+from collections import namedtuple
+from copy import copy
+
+
+_header = (('query', str),
+           ('subject', str),
+           ('percent_id', float),
+           ('aln_length', int),
+           ('mismatches', int),
+           ('gapopenings', int),
+           ('q_start', int),
+           ('q_end', int),
+           ('s_start', int),
+           ('s_end', int),
+           ('evalue', float),
+           ('bitscore', float))
+
+M9 = namedtuple('m9', [h[0] for h in _header])
+M9_empty = [M9(**{h: None for h, _ in _header})]
+
+
+def parse_m9(lines_or_fp):
+    """Parse m9 formatted tabular data
+
+    Parameters
+    ----------
+    lines : iterable of str, or a filepath
+        The lines to parse. It is expected that these lines are in BLAST m9
+        format, or comparable. Any additional columns will be ignored
+
+    Returns
+    -------
+    list of namedtuple
+        The namedtuples describe each field of the m9 format per record.
+    """
+    if isinstance(lines_or_fp, str):
```

Comment:
Was only being used via file object anyway so removed support for passing a
file path as thee isn't a need

On Thu, Apr 23, 2015 at 12:49 PM, Yoshiki Vázquez Baeza <
notifications@github.com> wrote:

> In platypus/parse.py
> <https://github.com/biocore/Platypus-Conquistador/pull/30#discussion_r28991623>
> :
>
> > +
> > +def parse_m9(lines_or_fp):
> > +    """Parse m9 formatted tabular data
> > +
> > +    Parameters
> > +    ----------
> > +    lines : iterable of str, or a filepath
> > +        The lines to parse. It is expected that these lines are in BLAST m9
> > +        format, or comparable. Any additional columns will be ignored
> > +
> > +    Returns
> > +    -------
> > +    list of namedtuple
> > +        The namedtuples describe each field of the m9 format per record.
> > +    """
> > +    if isinstance(lines_or_fp, str):
>
> Would you mind using skbio.io.util.open_file?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/Platypus-Conquistador/pull/30/files#r28991623>
> .
>